tests/spice_test.go

```diff
@@ -151,6 +152,35 @@ var _ = Describe("Vmlifecycle", func() {
 		}, 30)
 	})
 
+	Context("Two new VMs scheduled on the same node with a spice graphics device", func() {
+
+		It("should start without port clashes", func(done Done) {
+			// Create the VM
+			result := restClient.Post().Resource("vms").Namespace(tests.NamespaceTestDefault).Body(vm).Do()
+			obj, err := result.Get()
+			Expect(err).To(BeNil())
+
+			// Block until the VM is running
+			nodeName := tests.WaitForSuccessfulVMStart(obj)
+
+			vm1 := tests.NewRandomVMWithSpice()
+			vm1.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+			result = restClient.Post().Resource("vms").Namespace(tests.NamespaceTestDefault).Body(vm1).Do()
+			obj1, err := result.Get()
+			Expect(err).To(BeNil())
+
+			// Block until the VM is running
+			tests.WaitForSuccessfulVMStart(obj1)
+
+			obj, err = restClient.Get().Resource("vms").SetHeader("Accept", rest.MIME_JSON).SubResource("spice").Namespace(tests.NamespaceTestDefault).Name(vm.GetObjectMeta().GetName()).Do().Get()
+			Expect(err).To(BeNil())
+			obj1, err = restClient.Get().Resource("vms").SetHeader("Accept", rest.MIME_JSON).SubResource("spice").Namespace(tests.NamespaceTestDefault).Name(vm1.GetObjectMeta().GetName()).Do().Get()
+			Expect(err).To(BeNil())
+			Expect(obj.(*v1.Spice).Info.Port).To(BeNumerically("<", obj1.(*v1.Spice).Info.Port))
```

Comment:
personally I think that "less than" might be too specific here. yes, vm was created before vm1--so according to the current implementation this is true. If we ever change our mind and pick a slightly random port (e.g. to overcome race conditions), this isn't going to hold true.

All we really care about is "not equal", right?